datafusion/core/src/datasource/datasource.rs

```diff
@@ -72,3 +72,7 @@ pub trait TableProvider: Sync + Send {
         Ok(TableProviderFilterPushDown::Unsupported)
     }
 }
+
+pub trait TableProviderFactory: Sync + Send {
+    fn create(&self, name: &str, path: &str) -> Arc<dyn TableProvider>;
```

Comment:
Minor comment would be to call this `url` rather than `path`  to hint that the table provider factory could interpret it however it wanted and it wasn't necessarily a path to a local file system?